### PR TITLE
chore: Create script for generating COP23 test data

### PR DESCRIPTION
**Pull request by jason-p-pickering**
### Related Issues
- DP-716
- etc.

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

File: data-raw/COP23/generateCOP23TestData.R [hunk 1]
```diff
@@ -1,106 +1,152 @@
...
+                  value)
+
+
+
+  if (length(d$info$country_uids) == 1) {
+
+    d$datim$cop23_year2_sample_data  <- d$datim$cop23_sample_data %>%
+      dplyr::filter(attributeOptionCombo != datapackr::default_catOptCombo()) %>% #Remove IMPATT data
+      dplyr::group_by(dataElement, categoryOptionCombo) %>%
+      dplyr::summarise(value = sum(round(value * 1.25)), .groups = "drop") %>%
```
Comment (JordanBalesBAO) [inline data-raw/COP23/generateCOP23TestData.R:125]: Why 1.25?

File: data-raw/COP23/generateCOP23TestData.R [hunk 2]
```diff
@@ -228,56 +274,78 @@ sendDATIMExportToS3 <- function(d) {
...
 
-  # Load the file as a raw binary
-  read_file <- file(tmp, "rb")
-  raw_file <- readBin(read_file, "raw", n = file.size(tmp))
-  close(read_file)
+    object_tags <- createS3BucketTags(d)
 
-  object_tags <- createS3BucketTags(d)
+    object_name <- paste0("datim_export/", gsub("^20", "cop", d$info$cop_year),
+                          ifelse(d$info$cop_year == 2021, "_opu", ""),
```
Comment (JordanBalesBAO) [inline data-raw/COP23/generateCOP23TestData.R:305]: Is 2021 right here?

**Review by JordanBalesBAO — APPROVED** [review]
Functionally sound. Just random questions as it relates to naming and math.

File: data-raw/COP23/generateCOP23TestData.R (hunk 1 above)
Comment (jason-p-pickering) [reply data-raw/COP23/generateCOP23TestData.R:125]: Totally random choice really. I just multiplied the Year 1 data by this factor.

File: data-raw/COP23/generateCOP23TestData.R (hunk 2 above)
Comment (jason-p-pickering) [reply data-raw/COP23/generateCOP23TestData.R:305]: I copied and pasted some of this data from the app. Have removed these cop21 references, even though in this particular case, this code would not have had any effect.

**Pull request merged**